File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    openn = 0
    for i in s:
        if i == '(':
            openn += 1
        else:
            if openn > 0:
                openn -= 1
            else:
                answer = False
                break
    if openn != 0:
        answer = False
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    stack<char> ss;
    for(auto now : s){
        if(ss.empty()){
            if(now == '('){
                ss.push(now);
            }
            else{
                answer = false;
                break;
            }
        }else{
            if(now == '(') ss.push(now);
            else ss.pop();
        }
    }
    if(!ss.empty()) answer = false;

    return answer;
}